eessi_bot_event_handler.py

```diff
@@ -138,6 +284,72 @@ def handle_pull_request_event(self, event_info, log_file=None):
 
+    def handle_bot_command(self, event_info, bot_command, log_file=None):
+        """
+        Handle bot command
+
+        Args:
+            event_info (dict): object containing all information of the event
+            bot_command (EESSIBotCommand): command to be handled
+        """
+        cmd = bot_command.command
+        handler_name = f"handle_bot_command_{cmd}"
+        if hasattr(self, handler_name):
+            handler = getattr(self, handler_name)
+            self.log(f"Handling bot command {cmd}")
+            return handler(event_info, bot_command)
+        else:
+            self.log(f"No handler for command '{cmd}'")
+            raise EESSIBotCommandError(f"unknown command '{cmd}'; use `bot: help` for usage information")
+
+    def handle_bot_command_help(self, event_info, bot_command):
+        help_msg = "\n  **How to send commands to bot instances**"
+        help_msg += "\n  - Commands must be sent with a **new** comment (edits of existing comments are ignored)."
+        help_msg += "\n  - A comment may contain multiple commands, one per line."
+        help_msg += "\n  - Every command begins at the start of a line and has the syntax `bot: COMMAND [ARGUMENTS]*`"
+        help_msg += "\n  - Currently supported COMMANDs are: `help`, `build`, `showconfig`"
+        return help_msg
+
+    def handle_bot_command_build(self, event_info, bot_command):
+        gh = github.get_instance()
+        self.log("repository: '%s'", event_info['raw_request_body']['repository']['full_name'])
+        repo_name = event_info['raw_request_body']['repository']['full_name']
+        pr_number = event_info['raw_request_body']['issue']['number']
+        pr = gh.get_repo(repo_name).get_pull(pr_number)
+        build_msg = ''
+        if check_build_permission(pr, event_info):
+            # use filter from command
+            build_msg = submit_build_jobs(pr, event_info, bot_command.action_filters)
+        else:
+            request_body = event_info['raw_request_body']
+            sender = request_body['sender']['login']
+            build_msg = f"account {sender} has no permission to submit build jobs"
+        return build_msg
+
+#    def handle_bot_command_deploy(self, event_info, bot_command):
+#        return
+
+#    def handle_bot_command_enable(self, event_info, bot_command):
+#        return
+
+#    def handle_bot_command_disable(self, event_info, bot_command):
+#        return
+
+#    def handle_bot_command_cancel(self, event_info, bot_command):
+#        return
+
+#    def handle_bot_command_status(self, event_info, bot_command):
+#        return
+
+    def handle_bot_command_showconfig(self, event_info, bot_command):
+        self.log("processing bot command 'showconfig'")
+        gh = github.get_instance()
+        repo_name = event_info['raw_request_body']['repository']['full_name']
+        pr_number = event_info['raw_request_body']['issue']['number']
+        pr = gh.get_repo(repo_name).get_pull(pr_number)
```

Comment:
Wouldn't it make more sense to move this stuff to `handle_bot_command`, and pass down `pr` into `handle_bot_command_*`?

Reply:
In addition to `event_info` and `bot_command`? Maybe. Every method needs something slightly different. `_help` doesn't need any additional data, `_build` needs the `event_info` (for passing it down to existing functions) and `bot_command`. `_show_config` needs `event_info`. Both `_build` and `_show_config` use `event_info` to init an object for the pull request and pass that down to subsequent function calls. An improvement could be to put duplicate code into a function `get_pr_from_issue_event(event_info)` that returns the pull request object (we have to be a bit careful because different events have different data ... or at least name attributes differently).